### PR TITLE
Update GHA

### LINKED ISSUE #31: Linting, syntax checking
The repo needs some ci to lint and error check.

### LINKED ISSUE #32: Bump licence
It's 2022

### LINKED ISSUE #34: Pipeline Improvements
- [x] Builds should only happen when we publish a release on GitHub so we aren't building for every commit to `main`.
- [x] Linting and syntax checking should happen on push.

### PR DESCRIPTION
**Pull request by amberstarlight**
Changes the GHA workflows so we're not building a new image for every push to `main`. We will need to adopt semver and push tags when we are happy to do a build.

The pipeline is *going to fail* until I apply the output of eslint and prettier, which I will do in a separate PR so this one isn't bogged down.

Fixes #32 
Fixes #31
Fixes #34

### CODE REVIEW

File: .github/workflows/push.yml [hunk 1]
```diff
@@ -1,41 +1,19 @@
...
-        with:
-          context: .
-          push: true
-          tags: ghcr.io/cutie-club/cutiebot:latest
-          platforms: linux/amd64, linux/arm64, linux/arm/v7
-          cache-from: type=gha
-          cache-to: type=gha,mode=max
+      - run: yarn install
+      - run: yarn lint
+      - run: yarn pretty-check
```
Comment (mini-ninja-64) [inline .github/workflows/push.yml]: is pretty-check neccesary due to `eslint-plugin-prettier` not that theres any harm having both, just curious

**Review by mini-ninja-64 — CHANGES_REQUESTED** [review]
not really requesting change, just a thought

File: .github/workflows/push.yml (hunk 1 above)
Comment (amberstarlight) [reply .github/workflows/push.yml]: No you're right, running both is extraneous really. I'll remove `pretty-check` from the job.

**Review by mini-ninja-64 — APPROVED** [review]

**Pull request merged**